app/decorators/order_detail_decorator.rb

```diff
@@ -0,0 +1,6 @@
+class OrderDetailDecorator < SimpleDelegator
+
+  delegate :full_name, to: :assigned_user, prefix: :assigned_user, allow_nil: true
+  delegate :name, to: :order_status, prefix: :order_status, allow_nil: true
```

Comment:
Looking at OrderStatus, we might not even need this because it already defines to_s as name. So in HAML, = order_detail.order_status should render out to the status's name already and handle the nil case as well.